Content.Server/Store/Systems/StoreSystem.Ui.cs

```diff
@@ -97,9 +113,14 @@ public void UpdateUserInterface(EntityUid? user, EntityUid store, StoreComponent
         // TODO: if multiple users are supposed to be able to interact with a single BUI & see different
         // stores/listings, this needs to use session specific BUI states.
 
+        var xform = Transform(store);
+
+        if (component.StartingMap != xform.MapID)
+            component.RefundAllowed = false;
+
         // only tell operatives to lock their uplink if it can be locked
         var showFooter = HasComp<RingerUplinkComponent>(store);
-        var state = new StoreUpdateState(component.LastAvailableListings, allCurrency, showFooter);
+        var state = new StoreUpdateState(component.LastAvailableListings, allCurrency, showFooter, component.RefundAllowed);
```

Comment:
You shouldn't be updating the component instead of a UI state update i.e. if a client opens the UI it shouldn't change the component.

It should be moved elsewhere or just removed as a datafield. You also need to be validating it on purchase otherwise this will break, e.g. if they open the UI on Map 1, then move to Map 2 you never update this.